networktest.go

```diff
@@ -687,15 +687,70 @@ func (n *networkHarness) OpenChannel(ctx context.Context,
 
 	select {
 	case <-ctx.Done():
-		return nil, fmt.Errorf("timeout reached before chan pending " +
-			"update sent")
+		return nil, fmt.Errorf("timeout reached before chan pending "+
+			"update sent: %v", err)
 	case err := <-errChan:
 		return nil, err
 	case <-chanOpen:
 		return respStream, nil
 	}
 }
 
+// OpenPendingChannel attempts to open a channel between srcNode and destNode with the
+// passed channel funding parameters. If the passed context has a timeout, then
+// if the timeout is reached before the channel pending notification is
+// received, an error is returned.
+func (n *networkHarness) OpenPendingChannel(ctx context.Context,
+	srcNode, destNode *lightningNode, amt btcutil.Amount,
+	pushAmt btcutil.Amount, numConfs uint32) (*lnrpc.PendingUpdate, error) {
+
+	openReq := &lnrpc.OpenChannelRequest{
+		NodePubkey:         destNode.PubKey[:],
+		LocalFundingAmount: int64(amt),
+		PushSat:            int64(pushAmt),
+		NumConfs:           numConfs,
+	}
+
+	respStream, err := srcNode.OpenChannel(ctx, openReq)
+	if err != nil {
+		return nil, fmt.Errorf("unable to open channel between "+
+			"alice and bob: %v", err)
+	}
+
+	chanPending := make(chan struct{})
+	errChan := make(chan error)
+	var pendingUpdate *lnrpc.PendingUpdate
+	go func() {
+		// Consume the "channel pending" update. This waits until the node
+		// notifies us that the final message in the channel funding workflow
+		// has been sent to the remote node.
+		resp, err := respStream.Recv()
+		if err != nil {
+			errChan <- err
+			return
+		}
+		pendingResp, ok := resp.Update.(*lnrpc.OpenStatusUpdate_ChanPending)
+		if !ok {
+			errChan <- fmt.Errorf("expected channel pending update, "+
+				"instead got %v", resp)
+			return
+		}
+
+		pendingUpdate = pendingResp.ChanPending
```

Comment:
Minor nit, but why not just send the `ChanPending` response over the channel rather than assigning a variable, then closing a channel?